add --target flag examples (#8367)
fishtown-analytics/docs.getdbt.com: website/docs/reference/dbt-jinja-functions/target.md

| `target.project` | abc-123 | The project specified in the active profile |
| `target.dataset` | dbt_alice | The dataset the active profile |

## Using the --target flag

> The active target (and its associated `target.name` value) can be set using the `--target` flag when running dbt commands:

Reviewer: Suggest to use active voice:
```suggestion
Use the `--target` flag when running dbt commands to set the active target and its associated `target.name` value:
```

      {%- endif -%}
    schema: source_schema
```

> To specify which target to use when running dbt commands, use the `--target` flag:

Reviewer: I wonder if we should introduce `--target` earlier so readers understand how `target.name` is set before seeing the examples. Might be helpful to add a short section explaining how `--target` works before the Examples section.
